src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Service/AuthorizationService.php

```diff
@@ -226,6 +227,22 @@ function ($id) use ($contact) {
         );
     }
 
+    /**
+     * @param $serviceId
+     * @return \Surfnet\ServiceProviderDashboard\Domain\Entity\Service|null
+     * @throws ServiceNotFoundException
+     */
+    public function getServiceById($serviceId)
```

Comment:
A nice addition! Addidng additional verification logic is very usefull here.

Some thoughts:

1. The service can still hypothetically (very slim chance, as hasAccessToService already verified this. It is bad practice to count on this as that method might change or be replaced/removed) be null, as `return $this->serviceService->getServiceById($serviceId);` still uses the 'findById' method on the `ServiceService`. A null response here should also result in the `ServiceNotFoundException`. 
2. Calling `$this->setSelectedServiceId($serviceId);` should than be performed after getting the actual service.
3. The Exception message here can be made more verbose. I suggest somthing like: `Unable to find service with id %d`

Reply:
1. Changed
2. Changed
3. Made more verbose but without the parameter